ddtrace/contrib/pylons/middleware.py

```diff
@@ -48,6 +49,8 @@ def __call__(self, environ, start_response):
             # Set the service in tracer.trace() as priority sampling requires it to be
             # set as early as possible when different services share one single agent.
             span.span_type = http.TYPE
+            # Send to Trace Search. TODO: plug it to ddtrace.config.
+            span.set_tag(EVENT_SAMPLE_RATE_KEY, 1)
```

Comment:
I am open on whatever API we pick. I let you check with @palazzem which one you want to use.

Reply:
I talked with Manu about this. Let's keep as `set_tag` as it is a more future proof way of handling this that does not require us to add and then maintain another method for `Span` class

Reply:
I'd prefer to use the `span.set_tag()` because it's an API already available and we use it for priority sampling. To make Trace Search work OOTB (so a change in our defaults), I'd prefer to not extend the Tracer API unless really required.

`span. set_event_sample_rate(1.0)` is appealing (and also a right name from a design stand point) but it's a new API that will not work with other vendor neutral APIs such as OpenTracing. This limitation hits badly other languages where we only expose a Tracer/Span interface. For instance, in Go users should first type assert the interface to a concrete struct and then make the call. This results in an expensive operation (that is very expensive if done in a hot-path of users' app) and that will fail for other Tracer implementations.

Another option to avoid type assertion is to duplicate the behavior in a `span. set_event_sample_rate(1.0)` and also in `span.set_tag(EVENT_SAMPLE_RATE_KEY, 1.0)` but I think it's preferable to have one and only one way to define the event sampling rate.

Main thoughts are:

* let's change the default so that Trace Search is enabled by default
* let's provide a primitive to allow advanced usage (this API is used only with custom instrumentation)
* let's think if it's valuable to extend our API later